### DISCUSSION TITLE
WorkFlow Generalization

### DISCUSSION THREAD

**Discussion by backoffice01** [elsa-workflows/elsa-core#4182] [Q&A]
Hi, I am working on a requirement where we need to create dynamic workflows using Elsa programmatic workflow approach, like a single workflow class can create multiple workflows on the basis of some JSON configuration from database. is it possible that a signle class can generate multiple workflows with different names and different set of process?

**Comment by sfmskywalker** [comment]
Hey @backoffice01,

Absolutely, that's doable! You'd want to craft a new class that takes on the `IWorkflowProvider` role. With this in place, you can whip up multiple workflows, each with its own unique name and steps, all based on your JSON setup.
